includes/class-windows-azure-replace-media.php

```diff
@@ -213,77 +232,188 @@ private function is_image( $filetype ) {
-						// translators: %s would be an error message
-						printf( esc_html__( 'Error in uploading file. Error: %s', 'windows-azure-storage' ), esc_html( $e->getMessage() ) );
-					}
-				}
+			update_post_meta( $source_data['id'], '_wp_attachment_metadata', $source_data['meta_data'] );
+		}
 
-				update_post_meta( $source_attachment_id, 'windows_azure_storage_info', $source_azure_data );
+		if ( ! empty( $replace_data['meta_data']['sizes'] ) ) {
+			// Remove previous thumbnails 
+			// $this->delete_previous_thumbnails( $source_data );
+
+			// Let's replace the file remotely
+			foreach ( $sizes as $source_size => $sizes_source_data ) {
+				try {
+					\Windows_Azure_Helper::copy_media_to_blob_storage(
+						$this->container_name,
+						$sizes_source_data['replace_file'],
+						$sizes_source_data['source_file'],
+						$replace_data['mime_type'],
+						'',
+						'',
+						30,
+					);
+				} catch ( Exception $e ) {
+					// translators: %s would be an error message
+					printf( esc_html__( 'Error in uploading file. Error: %s', 'windows-azure-storage' ), esc_html( $e->getMessage() ) );
+				}
 			}
+		}
 
-			wp_delete_attachment( $media_to_replace_id, true );
+		wp_delete_attachment( $replace_data['id'], true );
+		
+		return $source_data;
+	}
+
+	/**
+	 * Delete blobs from source
+	 *
+	 * @param array $data source or replacement data
+	 * @return void
+	 */
+	public function delete_previous_thumbnails( $data ) {
+		$default_azure_storage_account_container_name = \Windows_Azure_Helper::get_default_container();
+
+		// delete remaining blobs from source
+		if ( ! empty( $data['meta_azure']['thumbnails'] ) ) {
+			foreach ( $data['meta_azure']['thumbnails'] as $blob_location ) {
+				try {
+					$full_blob_url = \Windows_Azure_Helper::get_full_blob_url( $blob_location );
+					if ( ! empty( $full_blob_url ) ) {
+						\Windows_Azure_Helper::delete_blob(
+							$default_azure_storage_account_container_name,
+							$blob_location,
+						);
+					}
+				} catch ( Exception $e ) {
+					// translators: %s would be an error message
+					printf( esc_html__( 'Blob could not be removed. Error: %s', 'windows-azure-storage' ), esc_html( $e->getMessage() ) );
+				}
+			}
 		}
+	}
 
-		$return_data['original_image']  = $source_filename;
-		$return_data['attachment_data'] = $source_attachment_data;
-		$return_data['attachment_data'] = $source_attachment_data;
-		$return_data['azure_data']      = $source_azure_data;
-		$return_data['version']         = $new_version;
+	/**
+	 * Get the nearest size from the replacement image
+	 *
+	 * @param array $source_sizes Source image data
+	 * @param array $target_sizes Replacement image data
+	 * @return array
+	 */
+	private function find_nearest_size( $source_sizes, $target_sizes ) {
+
+		$convert_sizes = array();
+		$filename_path = $source_sizes['meta_data']['file'];
+		$file_path     = dirname( $filename_path );
+
+		$target_filename  = $target_sizes['meta_data']['file'];
```

Comment:
Wondering if we need to be more defensive here? For both `$source_sizes` and `$target_sizes`, will these always be arrays and will the keys we want always exist? Or do we need to check for those before proceeding?